mock_context.go

```diff
@@ -0,0 +1,182 @@
+package fuego
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/go-fuego/fuego/internal"
+)
+
+// MockContext provides a framework-agnostic implementation of ContextWithBody
+// for testing purposes. It allows testing controllers without depending on
+// specific web frameworks like Gin or Echo.
+type MockContext[B any] struct {
+	internal.CommonContext[B]
+
+	RequestBody B
+	Headers     http.Header
+	PathParams  map[string]string
+	response    http.ResponseWriter
+	request     *http.Request
+	Cookies     map[string]*http.Cookie
+}
+
+// NewMockContext creates a new MockContext instance with the provided body
+func NewMockContext[B any](body B) *MockContext[B] {
+	return &MockContext[B]{
+		CommonContext: internal.CommonContext[B]{
+			CommonCtx:         context.Background(),
+			UrlValues:         make(url.Values),
+			OpenAPIParams:     make(map[string]internal.OpenAPIParam),
+			DefaultStatusCode: http.StatusOK,
+		},
+		RequestBody: body,
+		Headers:     make(http.Header),
+		PathParams:  make(map[string]string),
+		Cookies:     make(map[string]*http.Cookie),
+	}
+}
+
+// NewMockContextNoBody creates a new MockContext suitable for a request & controller with no body
+func NewMockContextNoBody() *MockContext[any] {
+	return NewMockContext[any](nil)
+}
+
+var _ ContextWithBody[string] = &MockContext[string]{}
+
+// Body returns the previously set body value
+func (m *MockContext[B]) Body() (B, error) {
+	return m.RequestBody, nil
+}
+
+// MustBody returns the body or panics if there's an error
+func (m *MockContext[B]) MustBody() B {
+	return m.RequestBody
+}
+
+// HasHeader checks if a header exists
+func (m *MockContext[B]) HasHeader(key string) bool {
+	_, exists := m.Headers[key]
+	return exists
+}
+
+// HasCookie checks if a cookie exists
+func (m *MockContext[B]) HasCookie(key string) bool {
+	_, exists := m.Cookies[key]
+	return exists
+}
+
+// Header returns the value of the specified header
+func (m *MockContext[B]) Header(key string) string {
+	return m.Headers.Get(key)
+}
+
+// SetHeader sets a header in the mock context
+func (m *MockContext[B]) SetHeader(key, value string) {
+	m.Headers.Set(key, value)
+}
+
+// PathParam returns a mock path parameter
+func (m *MockContext[B]) PathParam(name string) string {
+	return m.PathParams[name]
+}
+
+// Request returns the mock request
+func (m *MockContext[B]) Request() *http.Request {
+	return m.request
+}
+
+// Response returns the mock response writer
+func (m *MockContext[B]) Response() http.ResponseWriter {
+	return m.response
+}
+
+// SetStatus sets the response status code
+func (m *MockContext[B]) SetStatus(code int) {
+	if m.response != nil {
+		m.response.WriteHeader(code)
+	}
+}
+
+// Cookie returns a mock cookie
+func (m *MockContext[B]) Cookie(name string) (*http.Cookie, error) {
+	cookie, exists := m.Cookies[name]
```

Comment:
Cookie needs it OpenAPIParam input

Reply:
Why?

Reply:
Yeah it will be useful to set default values for example. Because defaults are defined in the route registration, not the controller.
But they'll be defined in 2 places : route registration & tests and it would be weird because nothing guarantees that it will be the same in these 2 different places.